src/storage/src/controller.rs

```diff
@@ -1156,91 +1156,116 @@ mod persist_read_handles {
                 let mut read_handles: BTreeMap<GlobalId, ReadHandle<SourceData, (), T, Diff>> =
                     BTreeMap::new();
 
-                while let Some(cmd) = rx.recv().await {
-                    // Peel off all available commands.
-                    // This allows us to catch up if we fall behind on downgrade commands.
-                    let mut commands = vec![cmd];
-                    while let Ok(cmd) = rx.try_recv() {
-                        commands.push(cmd);
-                    }
-                    // Collect all downgrade requests and apply them last.
-                    let mut downgrades = BTreeMap::default();
-                    let mut shutdown = false;
-
-                    for (span, command) in commands {
-                        match command {
-                            PersistWorkerCmd::Register(id, read_handle) => {
-                                let previous = read_handles.insert(id, read_handle);
-                                if previous.is_some() {
-                                    panic!(
-                                        "already registered a ReadHandle for collection {:?}",
-                                        id
-                                    );
-                                }
+                let mut shutdown = false;
+                let mut interval = tokio::time::interval(std::time::Duration::from_secs(60));
+                while !shutdown {
+                    tokio::select! {
+                        _ = interval.tick() => {
+                            // Ideally we would use the code below, but the `since`
+                            // lifetime expires to early for the future, or somesuch.
+
+                            // let futs = FuturesUnordered::new();
+                            // for (_id, read) in read_handles.iter_mut() {
+                            //     let since = read.since().clone();
+                            //     futs.push(read.maybe_downgrade_since(&since));
```

Comment:
The problem is that `maybe_downgrade_since` takes a `&Antichain` while `downgrade_since` takes an owned `Antichain`. The reasoning behind this is that `maybe_downgrade_since` is not sure whether it needs an owner version and only clones when necessary. @pH14 What do you think? Should we change `maybe_downgrade_since`? It would make it slightly less performant (if at all), but seems the win that we can get here is bigger.